crates/workspace/src/pane_group.rs

```diff
@@ -867,7 +867,14 @@ mod element {
             debug_assert!(flexes.len() == len);
             debug_assert!(flex_values_in_bounds(flexes.as_slice()));
 
-            let magnification_value = WorkspaceSettings::get(None, cx).active_pane_magnification;
+            let mut magnification_value =
```

Comment:
1. Let's do a single `WorkspaceSettings::get(None, cx)` call, instead 3 new calls below 
2. And ensure that the values are between `[0.0, 1.0]`, clipping them if otherwise. I also see below, that we ignore values if those are incorrect, we'd rather log something (if you want to) or take the closest possible value instead.

Reply:
> Let's do a single WorkspaceSettings::get(None, cx) call, instead 3 new calls below
after removing the check for the old parameter as suggested on the other comment, there's only one access inside this function, the other one is inside paint

> And ensure that the values are between [0.0, 1.0], clipping them if otherwise. I also see below, that we ignore values if those are incorrect, we'd rather log something (if you want to) or take the closest possible value instead.
magnification is not the feature I'm adding in this pr, that's already existing in zed and is not clamped, I can clamp it with a new commit if you want though, but the range wouldn't be `[0.0, 1.0]` as the main intent for this feature is to have values > 1, so maybe `[1.0, 10.0]`?

I did add clamping to the opacity though, as that was indeed missing